### PR TITLE
[Matrix][SYCL] Fix the bug of comparison between two bf16's wi_element

### PR DESCRIPTION
**Pull request by yubingex007-a11y**
Previously, as for bf16's comparison, we directly compare two unsigned short and it is wrong.
Now, we extend b16 to float and do the correct comparsion.

### CODE REVIEW

**Review by dkhaldi — DISMISSED** [review]
LGTM

**Comment by yubingex007-a11y** [comment]
@intel/llvm-reviewers-runtime ping

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp [hunk 1]
```diff
@@ -477,8 +477,8 @@ class wi_element<uint16_t, NumRows, NumCols, Layout, Group> {
 
   explicit operator bool() {
 #ifdef __SYCL_DEVICE_ONLY__
-    return __spirv_VectorExtractDynamic(M.spvm, idx) !=
-           static_cast<uint16_t>(0);
+    return std::fabs(make_fp32(__spirv_VectorExtractDynamic(M.spvm, idx))) <
```
Comment (yubingex007-a11y) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:480]: it seems element_wise_all_ops_bf16_yubing_20220128.cpp failed on this line.
when I change to
make_fp32(__spirv_VectorExtractDynamic(M.spvm, idx)) != 0.0f;
the testcase can pass

Comment (yubingex007-a11y) [reply sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp]: we use ">=" instead of "<"

**Comment by yubingex007-a11y** [comment]
ping? @dkhaldi

**Review by dkhaldi — APPROVED** [review]
LGTM

**Review by vladimirlaz — APPROVED** [review]

**Comment by yubingex007-a11y** [comment]
@vladimirlaz , hi, could you help us merge it ?

**Pull request merged**